csharp/test/Apache.Arrow.Tests/FieldComparer.cs

```diff
@@ -38,5 +39,26 @@ public static bool Equals(Field f1, Field f2)
             }
             return false;
         }
+
+        public static void Compare(Field expected, Field actual)
```

Comment:
Does this method do the same thing that `Equals` does above?

Reply:
They have the same logic, but different purposes.

This one Asserts that the 2 Fields are equal, by going through each property and `Assert.Equal`. When something is not equal, this is a better option because xunit will tell you exactly what and why doesn't equal. The above `Equals` method only returns true or false if the 2 Fields equal. I needed to leave the `Equals` method in because there are a few places that `Assert.False(FieldComparer.Equals(f1, f2))`. Those couldn't be migrated to the `Compare` method because we expect the two instances to be different - so the Asserts would throw.

Reply:
Ah ok. Sad that we need to have 2 copies for this. But, it's in testing code, so I'm not that concerned about it.